### PR TITLE
Enforce null comparisons to make XO happy

### PR DESCRIPTION
**Pull request by davegomez**
This pull request intents to remove the `react/no-eq-null` XO rule according to issue #730.

### CODE REVIEW

**Comment by rauchg** [comment]
I honestly love the loose comparison to both null and undefined :( Am I the only one?

**Comment by leo** [comment]
I don't hate them either. But we want to make XO happy!

**Comment by ekmartin** [comment]
`typeof config.prop !== 'undefined'` isn't necessary though (`config.prop !== undefined` would do the same). See my comment here: https://github.com/zeit/hyper/pull/760#issuecomment-250995605

File: package.json [hunk 1]
```diff
@@ -62,7 +62,6 @@
     ],
     "rules": {
       "eqeqeq": ["error", "allow-null"],
```
Comment (ekmartin) [inline package.json]: this line can also be removed

Comment (leo) [inline package.json]: @ekmartin The whole line 64? Should I push to master?

Comment (ekmartin) [inline package.json]: Yep, xo's `eqeqeq` is good if we don't want allow null.

Comment (leo) [inline package.json]: Done!

**Pull request merged**